### PR TITLE
coinph: add apis

### PR DESCRIPTION
**Pull request by sc0Vu**
In this PR, I added new apis in coinph.

TODO:

- [x] add fetchDepositAddress
- [x] update fetchDeposits
- [x] update fetchWithdrawals

### CODE REVIEW

File: ts/src/coinsph.ts [hunk 1]
```diff
@@ -1720,6 +1740,57 @@ export default class coinsph extends Exchange {
...
+        const networkCode = this.safeString (params, 'network');
+        const networkId = this.networkCodeToId (networkCode, code);
+        const currency = this.currency (code);
+        const request = {
+            'coin': currency['id'],
+        };
+        if (networkId !== undefined) {
+            request['network'] = networkId;
+        }
+        const response = await this.privateGetOpenapiWalletV1DepositAddress (this.extend (request, params));
```
Comment (carlosmiei) [inline ts/src/coinsph.ts:1764]: Hum I suspect `network` is required here
```
coinsph.fetchDepositAddress (USDT)
BadRequest coinsph {"code":-1130,"msg":"Parameter 'network' is not valid."}
---------------------------------------------------
[BadRequest] coinsph {"code":-1130,"msg":"Parameter 'network' is not valid."}

    at throwExactlyMatchedException  Users/cjg/Git/ccxt12/ccxt/js/src/base/Exchange.js:3038  
    at handleErrors                  Users/cjg/Git/ccxt12/ccxt/js/src/coinsph.js:1834        
    at                               Users/cjg/Git/ccxt12/ccxt/js/src/base/Exchange.js:744   
    at processTicksAndRejections     node:internal/process/task_queues:95                    
    at fetch2                        Users/cjg/Git/ccxt12/ccxt/js/src/base/Exchange.js:2631  
    at request                       Users/cjg/Git/ccxt12/ccxt/js/src/base/Exchange.js:2634  
    at fetchDepositAddress           Users/cjg/Git/ccxt12/ccxt/js/src/coinsph.js:1738        
    at async run                     Users/cjg/Git/ccxt12/ccxt/examples/js/cli.js:298        

coinsph {"code":-1130,"msg":"Parameter 'network' is not valid."}
```

File: ts/src/coinsph.ts [hunk 2]
```diff
@@ -1720,6 +1740,57 @@ export default class coinsph extends Exchange {
...
+         * @description fetch the deposit address for a currency associated with this account
+         * @see https://coins-docs.github.io/rest-api/#deposit-address-user_data
+         * @param {string} code unified currency code
+         * @param {object} params extra parameters specific to the bitget api endpoint
+         * @param {string} params.network network for fetch deposit address
+         * @returns {object} an [address structure]{@link https://docs.ccxt.com/#/?id=address-structure}
+         */
+        await this.loadMarkets ();
+        const networkCode = this.safeString (params, 'network');
+        const networkId = this.networkCodeToId (networkCode, code);
```
Comment (carlosmiei) [inline ts/src/coinsph.ts:1756]: @sc0Vu we need to map some networks, for instance 
this works:
`n coinsph fetchDepositAddress "USDT" '{"network":"ETH"}'`
but this does not:
` n coinsph fetchDepositAddress "USDT" '{"network":"ERC20"}'`

Comment (sc0Vu) [reply ts/src/coinsph.ts:1833]: @carlosmiei turns out, they don't have network like {ERC|TRC|BEP}20 (token spec name). Anyway I push some updates for this. Let me know if you find any issue.

Comment (carlosmiei) [reply ts/src/coinsph.ts:1833]: @sc0Vu thanks

**Comment by carlosmiei** [comment]
LGTM ✅ 
```
p coinsph fetchDepositAddress "USDT" '{"network":"ERC20"}'          
Python v3.10.9
CCXT v3.1.56
coinsph.fetchDepositAddress(USDT,{'network': 'ERC20'})
{'address': '0xcb0da',
 'currency': 'USDT',
 'info': {'address': '0xcda',
          'addressTag': '',
          'coin': 'USDT'},
 'network': None,
 'tag': None}
```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**